### PR TITLE
Update cilium ebpf for upstream license fix

### PR DESCRIPTION
**Pull request by dcmiddle**
Fixes: #131 

Newer version of ebpf removes unnecessary GPL licensed tool https://github.com/cilium/ebpf/commit/0f74f86fd36bf66feba3e61e16a880a51fd0d7e4

### CODE REVIEW

**Comment by mythi** [comment]
we may need to force all indirect `cilium/ebpf` dependencies to the desired version:
```
$ go mod edit -replace=github.com/cilium/ebpf=github.com/cilium/ebpf@v0.11.0
$ go mod tidy
$ go mod vendor
```

this is pulled via `containerd/cgroups/v2` which does not seem to have it updated...

File: src/shim/go.mod [hunk 1]
```diff
@@ -22,7 +22,7 @@ require (
 require (
 	github.com/Microsoft/go-winio v0.5.2 // indirect
 	github.com/Microsoft/hcsshim v0.9.3 // indirect
-	github.com/cilium/ebpf v0.9.1 // indirect
+	github.com/cilium/ebpf v0.11.0 // indirect
```
Comment (mythi) [inline src/shim/go.mod:25]: this one can be dropped.

Comment (dcmiddle) [reply src/shim/go.mod]: I may have misunderstood you. I tried dropping that line and CI choked...
``` 
bin/containerd-shim-rune-v2
go: inconsistent vendoring in /home/ecc/actions-runner/_work/enclave-cc/enclave-cc/src/github.com/confidential-containers/enclave-cc/src/shim:
	github.com/cilium/ebpf@v0.[11](https://github.com/confidential-containers/enclave-cc/actions/runs/6226114935/job/16897952955?pr=222#step:8:12).0: is marked as explicit in vendor/modules.txt, but not explicitly required in go.mod
```

Comment (mythi) [reply src/shim/go.mod]: I meant that the modification can be dropped and we keep v0.9.1 in go.mod. It's good to `go mod tidy && go mod vendor` after that.

**Review by mythi — APPROVED** [review]
thanks!

**Pull request merged**